### PR TITLE
#417 - preparing the backend for charts that contain vehicles and inv…

### PR DESCRIPTION
**Pull request by aviklai**
…olved data.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -169,6 +169,42 @@ def markers():
                                       is_thin),
                         mimetype="application/json")
 
+
+@app.route("/charts-data", methods=["GET"])
+@user_optional
+def charts_data():
+    logging.debug('getting charts data')
+    kwargs = get_kwargs()
+    accidents_vehicles_involved = Marker.bounding_box_query(is_thin=False, yield_per=50, involved_and_vehicles=True, **kwargs)
```
Comment (danielhers) [inline main.py]: you can do 

    accidents, vehicles, involved =... 

To get three lists and you won't need to use indices. 

Also, I think you can do these loops with list comprehension, just need to replace the dictionary element assignment by a function.

File: models.py [hunk 2]
```diff
@@ -333,7 +333,18 @@ def bounding_box_query(is_thin=False, yield_per=None, **kwargs):
...
+
+        if involved_and_vehicles:
+            markers_ids = []
+            for marker in markers:
+                markers_ids.append(marker.id)
+            markers = db_session.query(Marker).filter(Marker.id.in_(markers_ids))
+            vehicles = db_session.query(Vehicle).filter(Vehicle.accident_id.in_(markers_ids))
+            involved = db_session.query(Involved).filter(Involved.accident_id.in_(markers_ids))
+            all_data = (markers.all(), vehicles.all(), involved.all())
+            return all_data
```
Comment (danielhers) [inline models.py]: No need for the variable, just return the three lists with commas in-between (without parentheses)

File: models.py [hunk 3]
```diff
@@ -333,7 +333,18 @@ def bounding_box_query(is_thin=False, yield_per=None, **kwargs):
 
         if is_thin:
             markers = markers.options(load_only("id", "longitude", "latitude"))
-        return markers
+
+        if involved_and_vehicles:
+            markers_ids = []
+            for marker in markers:
```
Comment (danielhers) [inline models.py]: Use list comprehension for this

**Comment by aviklai** [comment]
@danielhers done.

**Pull request merged**